src/CSET/operators/transect.py

```diff
@@ -0,0 +1,172 @@
+    # Parse arguments in case running using CSET bake
+    if type(startxy) is not tuple:
+        startxy = tuple([float(i) for i in startxy.split(",")])
+    if type(endxy) is not tuple:
+        endxy = tuple([float(i) for i in endxy.split(",")])
+
+    # Find out xy coord name
+    y_name, x_name = get_cube_yxcoordname(cube)
+
+    lon_coord = cube.coord(x_name)
+    lat_coord = cube.coord(y_name)
+
+    _check_within_bounds(startxy, lat_coord, lon_coord)
+    _check_within_bounds(endxy, lat_coord, lon_coord)
+
+    # Compute vector distance between start and end points in degrees.
+    dist_deg = np.sqrt(((startxy[0] - endxy[0]) ** 2) + ((startxy[1] - endxy[1]) ** 2))
+
+    # For scenarios where coord is at 90 degree to the grid (i.e. no latitude/longitude change).
+    # Only xmin or ymin will be zero, not both (otherwise startxy and endxy the same).
+    if startxy[1] - endxy[1] == 0:
+        latslice_only = True
+    else:
+        latslice_only = False
+
+    if startxy[0] - endxy[0] == 0:
+        lonslice_only = True
+    else:
+        lonslice_only = False
+
+    # Compute minimum gap between x/y spatial coords.
+    xmin = np.min(lon_coord.points[1:] - lon_coord.points[:-1])
+    ymin = np.min(lat_coord.points[1:] - lat_coord.points[:-1])
+
+    # Depending on the transect angle relative to the grid
+    if latslice_only:
+        xpnts = np.repeat(startxy[1], int(dist_deg / ymin))
+        ypnts = np.linspace(startxy[0], endxy[0], int(dist_deg / ymin))
+        xaxis_coord = "latitude"
+    elif lonslice_only:
+        xpnts = np.linspace(startxy[1], endxy[1], int(dist_deg / xmin))
+        ypnts = np.repeat(startxy[0], int(dist_deg / xmin))
+        xaxis_coord = "longitude"
+    else:
+        # Else use the smallest grid space in x or y direction.
+        xpnts = np.linspace(startxy[1], endxy[1], int(dist_deg / np.min([xmin, ymin])))
+        ypnts = np.linspace(startxy[0], endxy[0], int(dist_deg / np.min([xmin, ymin])))
+
+        # If change in latitude larger than change in longitude:
+        if abs(startxy[0] - endxy[0]) > abs(startxy[1] - endxy[1]):
+            xaxis_coord = "latitude"
+        elif abs(startxy[0] - endxy[0]) <= abs(startxy[1] - endxy[1]):
+            xaxis_coord = "longitude"
+
+    # Create cubelist to store interpolated points along transect.
+    interpolated_cubes = iris.cube.CubeList()
+
+    # Iterate over all points along transect.
+    for i in range(0, xpnts.shape[0]):
+        logging.info("%s/%s", i + 1, xpnts.shape[0])
+
+        # Get point along transect.
+        cube_slice = cube.interpolate(
+            [(x_name, xpnts[i]), (y_name, ypnts[i])], iris.analysis.Linear()
+        )
+
+        if xaxis_coord == "latitude":
+            cube_slice.remove_coord(x_name)
+            cube_slice.remove_coord(y_name)
+            dist_coord = iris.coords.AuxCoord(
+                ypnts[i], long_name="latitude", units="degrees"
+            )
+            cube_slice.add_aux_coord(dist_coord)
```

Comment:
Should this be a dim_coord? If so, you can probably rearrange this if else to move all the common functionality outside the conditional.

Reply:
Changed to dim coord and moved repeated lines outside conditional to reduce number of lines